ui/packages/security-logout/src/main/webapp/js/main.js

```diff
@@ -31,47 +31,29 @@
   })
 
   $.get('../services/logout/actions', function(data) {
-    var actions = JSON.parse(data)
-    logout(actions).catch(error => {
+    var action = JSON.parse(data)
+    logout(action).catch(error => {
       console.log(error.message)
       $('#error-msg').show()
     })
   })
 })()
 
-async function logout(actions) {
-  let localLogoutAction
-  let externalLogoutActions = false
-  for (const action of actions) {
-    if (action.title === 'Local Logout') {
-      localLogoutAction = action
-    } else {
-      externalLogoutActions = true
-      // Certain protocols, e.g. SAML with the POST binding, use redirects that can't be followed
-      // with an AJAX request. For logout to work, we need to load the url in an iframe.
-      const iframe = document.createElement('iframe')
-      iframe.style.display = 'none'
-      iframe.src = action.url
-      document.body.appendChild(iframe)
-    }
-  }
-
-  // Wait 3 seconds for iframe logout
-  if (externalLogoutActions) {
-    await new Promise(r => setTimeout(r, 3000))
-  }
-
-  if (localLogoutAction) {
-    const response = await $.ajax(localLogoutAction.url)
-    if (response.mustCloseBrowser === true) {
-      $('#close-browser-msg').show()
-      return
-    }
+async function logout(action) {
+  const response = await $.ajax(action.url)
+  if (response.mustCloseBrowser === true) {
+    $('#close-browser-msg').show()
+    return
   }
 
   const queryParams = parseQueryParams()
   if (queryParams && queryParams['service']) {
     $('#link').attr('href', queryParams['service'])
+  } else if (queryParams && queryParams['prevurl']) {
+    $('#link').attr(
+      'href',
+      window.location.href.replace(/\/logout\/.*/, queryParams['prevurl'])
```

Comment:
❓ Why do you handle `prevurl` differently than `service`?

Reply:
The difference is whether you received an absolute url or not. I added `prevurl` because for example if an error occurs we'll get`/admin` as the prevurl.